### ISSUE TITLE
Improve function docstrings

### ISSUE THREAD

**Issue by rickstaa** [anuraghazra/github-readme-stats#2032]
**Is your feature request related to a problem? Please describe.**

Several docstrings in the codebase are either incomplete or non-existing. The following docstring for example, does not contain a description:

https://github.com/anuraghazra/github-readme-stats/blob/eacc3cd29cecde23d488023ff8dc2326bc69c25d/src/common/utils.js#L7-L28

and could therefore be improved to

```javascript
/** Renders error message on the card.
 * @param {string} message
 * @param {string} secondaryMessage
 * @returns {string}
 */
const renderError = (message, secondaryMessage = "") => {
  return `
    <svg width="495" height="120" viewBox="0 0 495 120" fill="none" xmlns="http://www.w3.org/2000/svg">
    <style>
    .text { font: 600 16px 'Segoe UI', Ubuntu, Sans-Serif; fill: #2F80ED }
    .small { font: 600 12px 'Segoe UI', Ubuntu, Sans-Serif; fill: #252525 }
    .gray { fill: #858585 }
    </style>
    <rect x="0.5" y="0.5" width="494" height="99%" rx="4.5" fill="#FFFEFE" stroke="#E4E2E2"/>
    <text x="25" y="45" class="text">Something went wrong! file an issue at https://tiny.one/readme-stats</text>
    <text data-testid="message" x="25" y="55" class="text small">
      <tspan x="25" dy="18">${encodeHTML(message)}</tspan>
      <tspan x="25" dy="18" class="gray">${secondaryMessage}</tspan>
    </text>
    </svg>
  `;
};
```

while others like 

https://github.com/anuraghazra/github-readme-stats/blob/eacc3cd29cecde23d488023ff8dc2326bc69c25d/src/cards/stats-card.js#L15-L47

don't contain a docstring at al

**Describe the solution you'd like**

I think it would increase the developer experience if these docstrings were updated/added.

**Comment by Jagrutiti** [comment]
Hey @rickstaa 

I would like to work on this one.

**Comment by Jagrutiti** [comment]
Could you tell me all the files that would need this update?

**Comment by rickstaa** [comment]
@Jagrutiti, first of all, welcome to the GRS community 🚀. Great that you want to help improve the codebase. This improvement request regarded all files in the `src` folder. I, however, understand that that might be a bit too much. Feel free to create a PR for as many as you want to take on 😄:+1:.

**Comment by vj-codes** [comment]
@rickstaa hey this issue seems to be inactive by the assigned person, can I create a PR for the ones I found? Thanks in advance!

**Comment by codingmickey** [comment]
Is this issue worked upon or does it need any help?

**Comment by rickstaa** [comment]
@Jagrutiti and @vj-codes can you please specify if you are still working on this issue and which files you are planning to tackle? Thanks in advance! 😄

**Comment by Jagrutiti** [comment]
Hey @rickstaa 

I have added a PR of the files I have worked on.

**Comment by rickstaa** [comment]
Pro-tip for people taking this issue on, [GitHub's co-pilot](https://docs.github.com/en/copilot/getting-started-with-github-copilot/getting-started-with-github-copilot-in-visual-studio-code) will make adding function docstrings very easy:

https://user-images.githubusercontent.com/17570430/193562723-003a2f7a-f0dd-4c51-a47a-2da23111a7b1.mp4

**Comment by Jagrutiti** [comment]
This is a great tip @rickstaa. I did it manually in Visual Studio. This looks easier

**Comment by rickstaa** [comment]
> This is a great tip @rickstaa. I did it manually in Visual Studio. This looks easier

It's an amazing tool it will most of the time auto-complete the code I need to write. Looks like we are getting replaced soon 😆.

**Comment by shivam1646** [comment]
Hey @rickstaa can I work on this for the remaining files?

**Comment by rickstaa** [comment]
Yea, of course. Great! I assigned you. Just create a pull request for the files you want to do.

**Comment by rym-oualha** [comment]
Is there still any help wanted ?

**Comment by rickstaa** [comment]
@rym-oualha Thanks for wanting to contribute to this repository. I already assigned @shivam1646, so it is best to wait till he creates a new pull request or states which files he is tackling.

**Comment by shivam1646** [comment]
Hey @rickstaa , I've created a PR of the files that I worked on.

**Comment by shivam1646** [comment]
#2129

**Comment by shivam1646** [comment]
Hey @rickstaa, just checking in if there are any updates for PR #2129

**Comment by rickstaa** [comment]
@shivam1646 I added the last docstrings and merged #2129 into the master. Thanks for your contribution!

**Comment by shivam1646** [comment]
Thank you @rickstaa
